### ISSUE TITLE
Test runner and mpp

### ISSUE THREAD

**Issue by 1Jajen1** [1Jajen1/propCheck#25]
Several things:
- A custom test runner with no dependencies other than kotlin mpp libraries (seperate module)
- Modules for interop with existing testing libraries
- As a step towards mpp: Drop all jvm dependencies

### Custom test runner:
Just a basic dsl to write property based tests that then will be detected and executed, reporting failure. This should be mpp if possible.

### Interop with testing libraries:
A custom test runner is all great and fine, but interop with other libraries is a good thing as well. As this means simply offering wrappers and dsl's to build tests with propCheck this should be a easy task. Just a module with the correct dependencies and wrappers around their methods/classes.

### Mpp:
There are only two jvm dependencies atm:
- arrow
- commons-math

`arrow` is going to get mpp support soon, no idea about `IO` tho, so it's best to wait on that for now.
`commons-math` is used in one place only and that is for statistics analysis. Either replace it with a pure kotlin lib or implement that method here. 

This is kept as one issue atm as a reminder. I'll split this soon when I have though about this some more. 
The interop libraries would be amazing to have for 1.0, a test runner would not be bad as well but can be put off. Mpp is not a priority until `arrow` has it's IO or something equivalent mpp compatible.
